charts/karpenter/templates/clusterrole.yaml

```diff
@@ -22,12 +22,15 @@ rules:
   - apiGroups: ["storage.k8s.io"]
     resources: ["storageclasses", "csinodes"]
     verbs: ["get", "watch", "list"]
-  - apiGroups: ["apps"]
-    resources: ["daemonsets"]
+  - apiGroups: ["apps", ""]
+    resources: ["daemonsets","replicasets", "replicationcontrollers"]
```

Comment:
Did we end up using these permissions?

Reply:
Yes, they're used to reduce our stabilization time (i.e. if there are no pending pods and daemonsets, replicasets, replicationcontrollers are all reporting the correct number of ready replicas then we can consider consolidating now as opposed to continuing to wait).

Reply:
Generally, the less coupled Karpenter is to higher level controllers the better.

Reply:
Could we track pod creation rates? e.g., if we haven't seen a pod created in n seconds, we can assume that all controllers are idle instead of checking specifically for a set of controllers.

Reply:
Otherwise we probably also want to add stateful sets to this list.

Reply:
Thanks, added statefulsets to what we check for.

Reply:
> Could we track pod creation rates? e.g., if we haven't seen a pod created in n seconds, we can assume that all controllers are idle instead of checking specifically for a set of controllers.

I'd love to get an answer for this.

Reply:
That's an alternative approach but I don't see it as being a better solution. A pause in pod creation doesn't mean that those pods are now ready and their requisite controllers healthy. There are trade-offs to both approaches.